pkg/discovery/cache/cache_test.go

```diff
@@ -36,9 +36,7 @@ func TestCacheAddresses(t *testing.T) {
 	}
 
 	got := c.Addresses()
-	sort.Slice(got, func(i, j int) bool {
```

Comment:
While this is a nice improvement, how it fixes any flakiness? That's exactly the same logic overall :thinking:

Reply:
Lol it should be `got[i] < `got[j]` if any :man_facepalming:

Reply:
The previousode didn't sort the slice at all, it was sorting the slice indices...

https://play.golang.org/p/G3nXNmUeNVU